### ISSUE TITLE
Storyblok added a custom attribute function to links in rich text, which allows users to customize the attribute field of the a tag, but the a tag is not rendered correctly

### ISSUE THREAD

**Issue by yanchenhao57** [storyblok/storyblok-js-client#464]
Storyblok added a **custom attribute** function to links in rich text on January 19, 2023, which allows users to customize the attribute field of the a tag, but the a tag is not rendered correctly。
When I add a custom attribute to the link in the rich text of storyblok,The data structure returned by storyblok is like this:

```json
{
    "type": "link",
    "attrs": {
        "href": "<https://login.yahoo.co.jp/>",
        "uuid": null,
        "anchor": null,
        "custom": {
            "rel": "nofollow",
            "title": "lsdlfasdfsddf"
        },
        "target": "_blank",
        "linktype": "url"
    }
}
```

It ends up rendering to the a tag on the page like this:

```html
<a href="https://login.yahoo.co.jp/" target="_blank" linktype="url" custom="[Object,Object]"><span class="">xxx</span></a>
```

User-defined attributes are not correctly placed in the a tag as attributes of the a tag，
In my opinion it should render like this:

```html
<a href="https://login.yahoo.co.jp/" target="_blank" linktype="url" rel="nofollow" title="lsdlfasdfsddf"><span class="">xxx</span></a>
```

I read the code for rendering rich text in storyblok-js-client and found the problem:

```tsx
// src/schema.ts

// ...

const link: MarkSchema = (node: ISbNode) => {
	const attrs = { ...node.attrs }
	const { linktype = 'url' } = node.attrs

	if (isEmailLinkType(linktype)) {
		attrs.href = `mailto:${attrs.href}`
	}

	if (attrs.anchor) {
		attrs.href = `${attrs.href}#${attrs.anchor}`
		delete attrs.anchor
	}

	return {
		tag: [
			{
				tag: 'a',
				attrs: attrs,
			},
		],
	}
}

// ...
```

I found that when processing a link, the above code did not correctly handle the custom field, which led to the problem I mentioned above.

I think it can be changed to the following, I don't know if it is suitable😆

**Comment by yanchenhao57** [comment]
```tsx
const link: MarkSchema = (node: ISbNode) => {
	const attrs = { ...node.attrs }
	const { linktype = 'url' } = node.attrs

	if (isEmailLinkType(linktype)) {
		attrs.href = `mailto:${attrs.href}`
	}

	if (attrs.anchor) {
		attrs.href = `${attrs.href}#${attrs.anchor}`
		delete attrs.anchor
	}

// +	if (attrs.custom) {
// +		for (const key in attrs.custom) {
// +		  attrs[key] = attrs.custom[key];
// +		}
// +		delete attrs.custom;
// +	  }

	return {
		tag: [
			{
				tag: 'a',
				attrs: attrs,
			},
		],
	}
}
```

**Comment by yanchenhao57** [comment]
Thank @thiagosaife

**Comment by ademarCardoso** [comment]
Hello @yanchenhao57 , this issue was fixed in the [version 5.6.0](https://github.com/storyblok/storyblok-js-client/releases/tag/v5.6.0)

Thanks for the feedback 😄
